util-core/src/main/scala/com/twitter/util/Function.scala

```diff
@@ -47,6 +47,38 @@ object Function {
    * Creates `() => A` function from given `Callable`.
    */
   def ofCallable[A](c: Callable[A]): () => A = () => c.call()
+
+  /**
+   * Creates a T => R from a JavaFunction. Used for easier interop
+   * between Java 8 and Twitter Util libraries.
+   */
+  def func[T, R](f: JavaFunction[T, R]) = new Function[T, R] {
+    override def apply(value: T): R = f(value)
+  }
+
+  /**
+   * Creates a T => Unit from a JavaConsumer.
+   * Useful for e.g. future.onSuccess
+   */
+  def cons[T](f: JavaConsumer[T]) = new Function[T, Unit] {
+    override def apply(value: T): Unit = f(value)
+  }
+
+  /**
+   * like func, but deals with checked exceptions as well
+   */
+  def exfunc[T, R](f: ExceptionalJavaFunction[T, R]) = new ExceptionalFunction[T, R] {
```

Comment:
Do you mean like "exFunc" and "exCons"? We could certainly do that. It conforms to the java/scala naming conventions. My only concern is that the camel casing itself might (possibly) look a (tiny) bit noisier than the plain lowercased versions:

```java
Future.value(3)
  .flatMap(exfunc(i -> ...))
  .map(exfunc(i -> ...)
  .onSuccess(excons(...));
```

or

```java
Future.value(3)
  .flatMap(exFunc(i -> ...))
  .map(exFunc(i -> ...))
  .onSuccess(exCons(..));
```

The intention is to let the lambda contents shine and hide all the noise, so to speak. With that argument, they could be shortened further:

```java
Future.value(3)
  .flatMap(xf(i -> ...))
  .map(xf(i -> ...))
  .onSuccess(xc(..));
```

with "f", "c", "xf" and "xc", but then again, that would probably make the code less understandable if you don't know the conventions.

Do you still want me to camelCase to exFunc and exCons?